### DISCUSSION TITLE
Presto code documentation for understanding its design and architecture

### DISCUSSION THREAD

**Discussion by acarpente-denodo** [prestodb/presto#23066] [Q&A] [answered]
Hi folks!

I'm new in the Presto community. I'm working on adding a new feature, and sometimes, I struggle to find helpful information about Presto code design that helps me understand its architecture.

I found [a paper on Trino's website](https://trino.io/Presto_SQL_on_Everything.pdf) that contains interesting information. I also read the book [Learning and Operating Presto](https://www.oreilly.com/library/view/learning-and-operating/9781098141844/).

Do you have any other documentation for developers I can read to learn more about Presto code design and architecture?

**Comment by aditi-pandit** [comment, marked as answer]
Hi @acarpente-denodo :

The Presto paper is a good resource for understanding Presto architecture.

I found [Presto: The Definitive Guide](https://www.amazon.com/Presto-Definitive-Guide-Storage-Environment/dp/149204427X) very useful when starting on Presto. [prestodb.io docs](https://prestodb.io/docs/current/index.html) were also are very helpful to get my head around the features and their outward functionality.

In terms of understanding code, I generally back-track from issues or other PRs in the area of the code/related features I'm interested in. Many issues are prefixed with [DESIGN] that can be looked at further e.g. [Aria optimization design](https://github.com/prestodb/presto/issues/16718). Tracing through code with examples also helps.

**Reply by acarpente-denodo** [reply]
Hi @aditi-pandit 

Thanks for sharing these useful resources. I appreciate you taking the time to look for this information. 

Thanks again for your help!
